src/aspire/utils/optimize.py

```diff
@@ -138,16 +138,15 @@ def identity(input_x):
         if cg_opt['verbose']:
             logger.info('[CG] Applying matrix & preconditioner')
 
-        a_p = a_fun(p)
-        old_gamma = np.real(np.sum(s.conj() * r, -1))
-
-        alpha = old_gamma / np.real(np.sum(p.conj() * a_p, -1))
+        a_p = np.array(a_fun(p), dtype=b.dtype)
+        old_gamma = np.sum(s.conj() * r, -1)
```

Comment:
Why is the cast to real gone here? Same for `alpha` and `new_gamma`.

Reply:
This is the original problem discussed for casting. I will put back the casting for alpha and gamma and make them consist with Matlab code. By the way, there have been no casts for Line 144 an 145 for  `x` and `a_x`which are consistent with Matlab version.

Reply:
It seems that casts in alpha and gamma need to be removed for complex matrix A as updates in new commit 6f43523.

Reply:
Why are they needed? The Matlab version works fine with complex matrix.

Reply:
Restored casts for alpha and gamma in commit 82cff7a to be consistent with Matlab version.